spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/transformer/VectorStoreRetriever.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.chat.prompt.transformer;
+
+import org.springframework.ai.chat.messages.Message;
+import org.springframework.ai.chat.messages.MessageType;
+import org.springframework.ai.document.Document;
+import org.springframework.ai.vectorstore.SearchRequest;
+import org.springframework.ai.vectorstore.VectorStore;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Transforms the PromptContext by retrieving documents from a VectorStore
+ */
+public class VectorStoreRetriever implements PromptTransformer {
+
+	private final VectorStore vectorStore;
+
+	private final SearchRequest searchRequest;
+
+	public VectorStoreRetriever(VectorStore vectorStore, SearchRequest searchRequest) {
```

Comment:
Was just kicking the tires on this PR and this constructor tripped me up a bit because it requires a `SearchRequest` as a parameter. 

My thinking was that I'd declare `ChatAgent` as a bean and then inject and use it in place of `ChatClient`. But you won't know how to create a `SearchRequest` at the time the application context is being created, so...

Is the intention that  instead of as a bean, the `ChatAgent` (specifically, the `DefaultChatAgent`) be created at the time that the query is being handled, because at that time you'd know how to build the `SearchRequest`?

Reply:
Thanks to @markpollack , I was made aware of `SearchRequest.defaults()` which lets me create a `VectorStoreRetriever` as a bean rather than in the controller. And, as Mark said, that same bean could use a `SearchRequest` with other filter info rather than the defaults, but I can still create it as a bean. So I withdraw my question.